src/app/clusters/dishwasher-alarm-server/dishwasher-alarm-server.cpp

```diff
@@ -0,0 +1,394 @@
+        return status;
+    }
+
+    ChipLogProgress(Zcl, "Dishwasher Alarm: Supported ep%d value: %" PRIx32 "", endpoint, supported.Raw());
+
+    // Whenever there is change in Supported attribute, Mask, State should change accordingly.
+    BitMask<AlarmMap> mask;
+    status = GetMaskValue(endpoint, &mask);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        return status;
+    }
+
+    if (!supported.HasAll(mask))
+    {
+        mask   = supported & mask;
+        status = SetMaskValue(endpoint, mask);
+    }
+    return status;
+}
+
+EmberAfStatus DishwasherAlarmServer::SetMaskValue(EndpointId endpoint, const BitMask<AlarmMap> mask)
+{
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+    status               = Attributes::Mask::Set(endpoint, mask);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Dishwasher Alarm: ERR: writing  mask, err:0x%x", status);
+        return status;
+    }
+
+    ChipLogProgress(Zcl, "Dishwasher Alarm: set Mask ep%d value: %" PRIx32 "", endpoint, mask.Raw());
+
+    // Whenever there is change in Mask, State should change accordingly.
+    BitMask<AlarmMap> state;
+    status = GetStateValue(endpoint, &state);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        return status;
+    }
+
+    if (!mask.HasAll(state))
+    {
+        state  = mask & state;
+        status = SetStateValue(endpoint, state);
+    }
+    return status;
+}
+
+EmberAfStatus DishwasherAlarmServer::SetLatchValue(EndpointId endpoint, const BitMask<AlarmMap> latch)
+{
+    if (!HasAlarmFeature(endpoint))
+    {
+        ChipLogProgress(Zcl, "Dishwasher Alarm feature: Unsupport Latch attribute");
+        return EMBER_ZCL_STATUS_UNSUPPORTED_ATTRIBUTE;
+    }
+
+    EmberAfStatus status = Attributes::Latch::Set(endpoint, latch);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Dishwasher Alarm: ERR: writing  latch, err:0x%x", status);
+        return status;
+    }
+
+    ChipLogProgress(Zcl, "Dishwasher Alarm: latch ep%d value: %" PRIx32 "", endpoint, latch.Raw());
+    return status;
+}
+
+EmberAfStatus DishwasherAlarmServer::SetStateValue(EndpointId endpoint, const BitMask<AlarmMap> newState)
+{
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+    BitMask<AlarmMap> currentState;
+
+    status = Attributes::State::Get(endpoint, &currentState);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Dishwasher Alarm: ERR: reading  state, err:0x%x", status);
+        return status;
+    }
+
+    status = Attributes::State::Set(endpoint, newState);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Dishwasher Alarm: ERR: writing  state, err:0x%x", status);
+        return status;
+    }
+
+    ChipLogProgress(Zcl, "Dishwasher Alarm: State ep%d value: %" PRIx32 "", endpoint, newState.Raw());
+
+    // If the feature is true, the latch operation can only be performed
```

Comment:
I'm sorry; I don't understand either this comment or this code.

SetStateValue should accept any value that's a subset of Supported.  Calling this function should not affect Latch in any way.  Why do you think it should affect it?